### PR TITLE
MF-507 change of circumstances

### PR DESCRIPTION
**Pull request by scottywakefield**
some acceptance tests are still ignored. :(

### CODE REVIEW

File: src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Services/WhenChangingPartyOfAReservation.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                .Setup(repository => repository.GetById(request.ReservationId))
+                .ReturnsAsync((Domain.Entities.Reservation) null);
+
+            var act = new Func<Task>(async () => await service.ChangeOfParty(request));
+
+            act.Should().Throw<EntityNotFoundException<Domain.Entities.Reservation>>();
+        }
+
+        [Test, RecursiveMoqAutoData]
+        public void And_Reservation_Not_Confirmed_Status_Then_Throws_ArgumentException(
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Services/WhenChangingPartyOfAReservation.cs:33]: Do we need a similar test to cover the scenario of making sure that we can't clone a reservation with a deleted status?

File: src/SFA.DAS.Reservations.Application/AccountReservations/Services/AccountReservationService.cs [hunk 2]
```diff
@@ -102,6 +110,58 @@ public async Task DeleteReservation(Guid reservationId)
...
+        public async Task<Guid> ChangeOfParty(ChangeOfPartyServiceRequest request)
+        {
+            var existingReservation = await _reservationRepository.GetById(request.ReservationId);
+            if (existingReservation == null)
+            {
+                throw new EntityNotFoundException<Domain.Entities.Reservation>();
+            }
+
+            if (existingReservation.Status != (short)ReservationStatus.Confirmed &&
+                existingReservation.Status != (short)ReservationStatus.Change)
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Application/AccountReservations/Services/AccountReservationService.cs:122]: The existing Reservation status can be **Change**, you should be able to go from employer a->b->c->n, each time calling change will look at the reservation passed. The only statuses that should be ignored are Completed and Deleted

File: src/SFA.DAS.Reservations.Api.AcceptanceTests/Features/ChangeOfParty.feature [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	| 1                       | 15214       | 2019-07-01 | 2019-01-01   | 1         | Confirmed | False           |
+	And I want to change account legal entity to 20
+	When I call change of party
+	Then an http status code of 200 is returned
+	And I have the following reservations:
+	| Account Legal Entity Id | Provider Id | Start Date | Created Date | Course Id | Status    | Is Levy Account |
+	| 1                       | 15214       | 2019-07-01 | 2019-01-01   | 1         | Confirmed | False           |
+	| 20                      | 15214       | 2019-07-01 | today        | 1         | Change    | True            |
+
+Scenario: Change employer, levy to non-levy - not supported
```
[the remark this answers is not in the document]
Comment (scottywakefield) [reply src/SFA.DAS.Reservations.Api.AcceptanceTests/Features/ChangeOfParty.feature]: removed the erroneous note

File: src/SFA.DAS.Reservations.Api/Controllers/ReservationsController.cs [hunk 4]
```diff
@@ -281,5 +281,50 @@ public async Task<IActionResult> BulkCreate([FromRoute]long accountLegalEntityId
...
+            {
+                var result = await _mediator.Send(new ChangeOfPartyCommand
+                {
+                    ReservationId = id,
+                    AccountLegalEntityId = request.AccountLegalEntityId,
+                    ProviderId = request.ProviderId
+                });
+                return Ok(new ChangeOfPartyResponse
+                {
+                    ReservationId = result.ReservationId
```
[the remark this answers is not in the document]
Comment (scottywakefield) [reply src/SFA.DAS.Reservations.Api/Controllers/ReservationsController.cs]: done

File: src/SFA.DAS.Reservations.Application/AccountReservations/Services/AccountReservationService.cs (hunk 2 above)
Comment (scottywakefield) [reply src/SFA.DAS.Reservations.Application/AccountReservations/Services/AccountReservationService.cs:122]: supported

File: src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Services/WhenChangingPartyOfAReservation.cs (hunk 1 above)
Comment (scottywakefield) [reply src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Services/WhenChangingPartyOfAReservation.cs]: done

**Pull request merged**